src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -0,0 +1,480 @@
+        {
+            send504(request);
+            canHandleRequest = true;
+        }
+
+        return canHandleRequest;
+    }
+
+    public void sendPendingInitialRequests(
+        int requestHash)
+    {
+        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.remove(requestHash);
+        if (pendingInitialRequests != null)
+        {
+            final PendingInitialRequests newPendingInitialRequests = pendingInitialRequests.withNextInitialRequest();
+            if (newPendingInitialRequests != null)
+            {
+                pendingInitialRequestsMap.put(requestHash, newPendingInitialRequests);
+                sendPendingInitialRequest(newPendingInitialRequests.initialRequest());
+            }
+        }
+    }
+
+    public void send304ToPendingInitialRequests(
+        int requestHash)
+    {
+        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.remove(requestHash);
+        if (pendingInitialRequests != null)
+        {
+            DefaultCacheEntry cacheEntry = cachedEntries.get(requestHash);
+            pendingInitialRequests.subcribers().forEach(request ->
+            {
+                send304(cacheEntry, request);
+                request.purge();
+            });
+            pendingInitialRequests.removeAllSubscribers();
+        }
+    }
+
+    public void removeAllPendingInitialRequests(
+        int requestHash)
+    {
+        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.remove(requestHash);
+        if (pendingInitialRequests != null)
+        {
+            pendingInitialRequests.removeAllSubscribers();
+        }
+    }
+
+    public void addPendingRequest(
+        HttpCacheProxyCacheableRequest initialRequest)
+    {
+        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.get(initialRequest.requestHash());
+        pendingInitialRequests.subscribe(initialRequest);
+    }
+
+    public void createPendingInitialRequests(
+        HttpCacheProxyCacheableRequest initialRequest)
+    {
+        pendingInitialRequestsMap.put(initialRequest.requestHash(), new PendingInitialRequests(initialRequest));
+    }
+
+    public void removePendingInitialRequest(
+        HttpCacheProxyCacheableRequest request)
+    {
+        PendingInitialRequests pendingInitialRequests = pendingInitialRequestsMap.get(request.requestHash());
+        if (pendingInitialRequests != null)
+        {
+            pendingInitialRequests.removeSubscriber(request);
+            if(pendingInitialRequests.numberOfSubscribers() == 0)
+            {
+                pendingInitialRequestsMap.remove(request.requestHash());
+            }
+        }
+    }
+
+    public void serveNextPendingInitialRequest(HttpCacheProxyCacheableRequest request)
+    {
+        removePendingInitialRequest(request);
+        sendPendingInitialRequests(request.requestHash());
+    }
```

Comment:
Let's remove these 4 methods and move to a shared state design, stored closer to the `HttpProxyCacheableRequest`.

Let's have a map keyed by `requestHash` to `HttpProxyCacheableRequestGroup`.

Then on `HttpProxyCacheableRequestGroup` we can store the set of `replyId` values for all the requests in the same group. Each `HttpProxyCacheableRequest` is responsible for adding and remove themselves from the set, in case they are aborted before completion for example.

When `HttpProxyCacheableRequest.newResponse` is invoked, we'll know which stream it is working on behalf of and also the group of all streams that would benefit from the same response.

If the response is cacheable, then we can signal to the other responses when it is safe to share the response, perhaps immediately (ideal).

If the response is _not_ cacheable, then we can select a different stream in the same group, and signal them to send their request on behalf of the group instead.